extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultCredentialsProvider.java

```diff
@@ -3,28 +3,29 @@
 import java.util.HashMap;
 import java.util.Map;
 
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
 import io.quarkus.credentials.CredentialsProvider;
 import io.quarkus.vault.VaultException;
+import io.quarkus.vault.VaultKVSecretEngine;
 import io.quarkus.vault.runtime.config.CredentialsProviderConfig;
 import io.quarkus.vault.runtime.config.VaultRuntimeConfig;
 
+@ApplicationScoped
+@Named("vault-credentials-provider")
```

Comment:
Why the `@Named` qualifier? Is it required for some functionality?

Reply:
it is related to https://quarkus.io/guides/credentials-provider#new-credentials-provider-extension
it is needed if you have more than 1 credentials provider in your app.

Reply:
Hm, are all credential providers required to be `@Named`? I'm a bit puzzled because the `MyCredentialsProvider` in https://quarkus.io/guides/credentials-provider#custom-credentials-provider is not `@Named`.

Reply:
> are all credential providers required to be @Named

no, if you create your own credentials provider, and do not include any other provider (e.g. the vault credentials provider) in your app. but if plan to include several providers in your app (hashicorp vault, plus your own provider or any other provider implemented by an extension - e.g. an azure kms), then you will need to distinguish them by name. in the doc it says in the section about providers implemented by extensions: _First, you need to name it to avoid collisions in case multiple credentials providers are available in the project:_
if you create your own provider, then you probably know if there will be multiple providers in your app. but if you create an extension with a provider in it, then you don't know what the extension's users will do with it.

it is the credentials provider's client (e.g. agroal) responsibility to select the appropriate provider, by calling the common service `io.quarkus.credentials.runtime.CredentialsProviderFinder.find(String)`
for instance in agroal `Datasources`:
```
        // credentials provider
        if (dataSourceRuntimeConfig.credentialsProvider.isPresent()) {
            String beanName = dataSourceRuntimeConfig.credentialsProviderName.orElse(null);
            CredentialsProvider credentialsProvider = CredentialsProviderFinder.find(beanName);
            String name = dataSourceRuntimeConfig.credentialsProvider.get();
            connectionFactoryConfiguration
                    .credential(new AgroalVaultCredentialsProviderPassword(name, credentialsProvider));
        }
```